### PR TITLE
Adding the Huber Loss function

### PR DESCRIPTION
**Pull request by atreyamaj**
## Description
Added the Huber loss function.  Referencing Issue #2478 

Fix #(issue)
Added code for the Huber loss function in `models/losses.py`. Also added relevant tests for both pytorch and tensorflow in `models/tests/test_losses.py`.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

Mypy and flake8 show errors however the errors are not in the code I added. They were present earlier.

@rbharath  @alat-rights

### CODE REVIEW

**Comment by alat-rights** [comment]
It looks like this only includes the code for the tests...

Did you run `git add` on `losses.py` too?

**Comment by atreyamaj** [comment]
> It looks like this only includes the code for the tests...
> 
> Did you run `git add` on `losses.py` too?

My apologies @alat-rights , I have added `losses.py` as well as made some more formatting changes (yapf).

**Comment by alat-rights** [comment]
Thank you! 

I only had one comment but I'm not sure if you *need* to address it, since your documentation seems similar to the other classes in the file.

Regardless, I'm sure a tiny bit more detail wouldn't hurt. 

Thanks for the contribution! This looks good to me.

Let's wait to see what the reviewers think 😃

**Comment by atreyamaj** [comment]
> Thank you!
> 
> I only had one comment but I'm not sure if you _need_ to address it, since your documentation seems similar to the other classes in the file.
> 
> Regardless, I'm sure a tiny bit more detail wouldn't hurt.
> 
> Thanks for the contribution! This looks good to me.
> 
> Let's wait to see what the reviewers think

Please let me know the change/comment, I'll incorporate it now!

**Comment by alat-rights** [comment]
You should be able to see the comment on GitHub? It should say "Alat-rights reviewed these changes-- 1 Comment" just above my message. If you click on "1 comment" you can see my comment.

It was basically that I think you could've added some comments to the methods to describe what they do, and possibly been a little more detailed with the comment describing the class.

**Comment by atreyamaj** [comment]
> You should be able to see the comment on GitHub? It should say "Alat-rights reviewed these changes-- 1 Comment" just above my message. If you click on "1 comment" you can see my comment.
> 
> It was basically that I think you could've added some comments to the methods to describe what they do, and possibly been a little more detailed with the comment describing the class.



> You should be able to see the comment on GitHub? It should say "Alat-rights reviewed these changes-- 1 Comment" just above my message. If you click on "1 comment" you can see my comment.
> 
> It was basically that I think you could've added some comments to the methods to describe what they do, and possibly been a little more detailed with the comment describing the class.

I don't think a review was published :thinking: 
I've added a bit more detail in the multi-line comment for the HuberLoss class about where it's used/useful and the default parameters.

**Comment by alat-rights** [comment]
It looks good to me! Let's wait for more comments from the reviewers.

File: deepchem/models/losses.py [hunk 1]
```diff
@@ -41,6 +41,23 @@ def _create_pytorch_loss(self):
     return torch.nn.L1Loss(reduction='none')
 
 
+class HuberLoss(Loss):
```
Comment (rbharath) [inline deepchem/models/losses.py:44]: Could you add the new loss to the docs? 

https://github.com/deepchem/deepchem/blob/master/docs/source/api_reference/models.rst#losses

**Review by rbharath — COMMENTED** [review]
Looks pretty good! Need to add the loss to the docs and I think it should be good on my end

File: deepchem/models/losses.py (hunk 1 above)
Comment (atreyamaj) [reply deepchem/models/losses.py:44]: @rbharath  Thank you! I have made the change.

**Comment by peastman** [comment]
Looks good!  Yapf is complaining though.  Did you use a different version from 0.22?

**Comment by atreyamaj** [comment]
> Looks good! Yapf is complaining though. Did you use a different version from 0.22?

Yes I believe I used 0.31
Should I roll back to 0.22 and run it again? 

@peastman

**Comment by peastman** [comment]
Yes, we're still using 0.22, so CI will fail if you use a different version.

**Comment by atreyamaj** [comment]
> Yes, we're still using 0.22, so CI will fail if you use a different version.

I rolled back yapf to 0.22 and ran it again just now, it says no changes @peastman

**Comment by peastman** [comment]
The errors are in test_gcn.py.

**Comment by atreyamaj** [comment]
> The errors are in test_gcn.py.

Thank you, I have pushed the new change now.

**Comment by rbharath** [comment]
`yapf` looks to be passing now and the only failures have been fixed in other branches (the mol2vec issue), so I'll go ahead and merge this in
